superset-frontend/src/SqlLab/components/ResultSet/ResultSet.test.tsx

```diff
@@ -0,0 +1,212 @@
+    id: 'ryhHUZCGb',
+    progress: 100,
+    state: 'fetching',
+    startDttm: Date.now() - 500,
+  },
+};
+const cachedQueryProps = { ...mockedProps, query: cachedQuery };
+const failedQueryWithErrorMessageProps = {
+  ...mockedProps,
+  query: failedQueryWithErrorMessage,
+};
+const failedQueryWithErrorsProps = {
+  ...mockedProps,
+  query: failedQueryWithErrors,
+};
+const newProps = {
+  query: {
+    cached: false,
+    resultsKey: 'new key',
+    results: {
+      data: [{ a: 1 }],
+    },
+  },
+};
+fetchMock.get('glob:*/api/v1/dataset?*', { result: [] });
+
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+const setup = (props?: any, store?: Store) =>
+  render(<ResultSet {...props} />, {
+    useRedux: true,
+    ...(store && { store }),
+  });
+
+describe('ResultSet', () => {
+  it('renders a Table', async () => {
+    const { getByTestId } = setup(mockedProps, mockStore(initialState));
+    const table = getByTestId('table-container');
+    expect(table).toBeInTheDocument();
+  });
+
+  it('should render success query', async () => {
+    const { queryAllByText, getByTestId } = setup(
+      mockedProps,
+      mockStore(initialState),
+    );
+
+    const table = getByTestId('table-container');
+    expect(table).toBeInTheDocument();
+
+    const firstColumn = queryAllByText(
+      mockedProps.query.results?.columns[0].name ?? '',
+    )[0];
+    const secondColumn = queryAllByText(
+      mockedProps.query.results?.columns[1].name ?? '',
+    )[0];
+    expect(firstColumn).toBeInTheDocument();
+    expect(secondColumn).toBeInTheDocument();
+
+    const exploreButton = getByTestId('explore-results-button');
+    expect(exploreButton).toBeInTheDocument();
+  });
+
+  it('should render empty results', async () => {
+    const props = {
+      ...mockedProps,
+      query: { ...mockedProps.query, results: { data: [] } },
+    };
+    const { getByRole } = setup(props, mockStore(initialState));
+
+    const alert = getByRole('alert');
+    expect(alert).toBeInTheDocument();
+    expect(alert).toHaveTextContent('The query returned no data');
+  });
+
+  it('should call reRunQuery if timed out', async () => {
+    const store = mockStore(initialState);
+    const propsWithError = {
+      ...mockedProps,
+      query: { ...queries[0], errorMessage: 'Your session timed out' },
+    };
+
+    setup(propsWithError, store);
+    expect(store.getActions()).toHaveLength(1);
+    expect(store.getActions()[0].query.errorMessage).toEqual(
+      'Your session timed out',
+    );
+    expect(store.getActions()[0].type).toEqual('START_QUERY');
+  });
+
+  it('should not call reRunQuery if no error', async () => {
+    const store = mockStore(initialState);
+    setup(mockedProps, store);
+    expect(store.getActions()).toEqual([]);
+  });
+
+  it('should render cached query', async () => {
+    const store = mockStore(initialState);
+    const { rerender } = setup(cachedQueryProps, store);
+
+    // @ts-ignore
```

Comment:
What is the TS error here?

Reply:
There're a lot of missing properties in User, ResultSet and so on
It need a huge refactoring in `src/SqlLab/fixtures`
I think this PR is big and we need to do it in next PR

Reply:
I agree, that would diverge a bit from this original PR. I think this is okay for now, if you open up a PR to fix this typing issue feel free to tag me and I'd be happy to review it 😁